scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/PlanWindow.java

```diff
@@ -0,0 +1,167 @@
+    private HLayout yesNoButtons;
+
+    private IButton yesButton;
+
+    private IButton noButton;
+
+    /**
+     * Default constructor
+     *
+     * @param controller
+     */
+    public PlanWindow(SchedulerController controller) {
+        this.controller = controller;
+        this.build();
+    }
+
+    /**
+     * Shows the window created by the constructor
+     */
+    public void show() {
+        this.window.show();
+    }
+
+    /**
+     * Destroy the window, you may null the reference after this as it will not
+     * be usable again
+     */
+    public void destroy() {
+        this.window.destroy();
+    }
+
+    private void initRootPage() {
+        rootPage = new VLayout();
+        rootPage.setMargin(20);
+        rootPage.setWidth100();
+        rootPage.setHeight100();
+    }
+
+    private void initMessagesPart() {
+        VLayout messagePanel = new VLayout();
+
+        Label confirmationLabel = new Label("You will open a new tab with the Job Planner Portal. Do you confirm?");
+        confirmationLabel.setHeight(30);
+        confirmationLabel.setWidth100();
+        confirmationLabel.setMargin(10);
+        confirmationLabel.setAlign(Alignment.CENTER);
+
+        messagePanel.addMember(confirmationLabel);
+        rootPage.addMember(messagePanel);
+    }
+
+    private void initButtonsPart() {
+        yesNoButtons = new HLayout();
+        yesNoButtons.setMargin(10);
+        yesNoButtons.setMembersMargin(5);
+        yesNoButtons.setHeight(30);
+        yesNoButtons.setWidth100();
+        yesNoButtons.setAlign(Alignment.RIGHT);
+
+        yesButton = new IButton("Yes");
+        yesButton.addClickHandler(new ClickHandler() {
+            @Override
+            public void onClick(ClickEvent event) {
+                String url = "/automation-dashboard/#/portal/job-planner-calendar-def-workflows";
+                com.google.gwt.user.client.Window.open(url, url, "");
+                PlanWindow.this.window.hide();
+                PlanWindow.this.destroy();
+            }
+        });
+
+        noButton = new IButton("No");
+        noButton.addClickHandler(new ClickHandler() {
+            @Override
+            public void onClick(ClickEvent event) {
+                PlanWindow.this.window.hide();
+                PlanWindow.this.destroy();
+            }
+        });
+
+        yesNoButtons.setMembers(yesButton, noButton);
+        rootPage.addMember(yesNoButtons);
+    }
+
+    private void build() {
+
+        initRootPage(); // ------------ root page of the window
+        initMessagesPart(); // -------- Select workflow Panel
+        initButtonsPart(); // ------------ Fill workflow variables Panel
+
+        this.window = new Window();
+        this.window.setTitle("Plan a job");
+        this.window.setShowMinimizeButton(false);
+        this.window.setIsModal(true);
+        this.window.setShowModalMask(true);
+        this.window.addItem(rootPage);
+        this.window.setWidth(this.width);
```

Comment:
The `this` of `this.width` should be removed since width is a constant.
The same goes for the `height` below.